src/dsp_tools/commands/validate_data/make_data_rdf.py

```diff
@@ -22,208 +27,147 @@
+        literal_value = Literal(val.object_value, datatype=prop_type_info.xsd_type)
+    else:
+        literal_value = Literal("", datatype=XSD.string)
+    g.add((val_iri, prop_type_info.knora_prop, literal_value))
+    g.add((res_iri, URIRef(val.prop_name), val_iri))
+    return g
+
+
+def _transform_link_value(val: ValueDeserialised, res_iri: URIRef) -> Graph:
+    object_value = val.object_value if val.object_value is not None else ""
+    g = Graph()
+    val_iri = DATA[str(uuid4())]
+    g.add((val_iri, RDF.type, KNORA_API.LinkValue))
+    g.add((val_iri, API_SHAPES.linkValueHasTargetID, DATA[object_value]))
+    g.add((res_iri, URIRef(val.prop_name), val_iri))
+    return g
+
+
+def _transform_list_value(val: ListValueDeserialised, res_iri: URIRef) -> Graph:
     node_name = val.object_value if val.object_value is not None else ""
-    return ListValueRDF(
-        prop_name=URIRef(val.prop_name),
-        object_value=Literal(node_name, datatype=XSD.string),
-        list_name=Literal(val.list_name, datatype=XSD.string),
-        res_iri=res_iri,
-    )
-
-
-def _transform_time_value(val: ValueDeserialised, res_iri: URIRef) -> ValueRDF:
-    content = (
-        Literal(val.object_value, datatype=XSD.dateTimeStamp)
-        if val.object_value is not None
-        else Literal("", datatype=XSD.string)
-    )
-    return TimeValueRDF(URIRef(val.prop_name), content, res_iri)
-
-
-def _transform_uri_value(val: ValueDeserialised, res_iri: URIRef) -> ValueRDF:
-    content = (
-        Literal(val.object_value, datatype=XSD.anyURI)
-        if val.object_value is not None
-        else Literal("", datatype=XSD.string)
-    )
-    return UriValueRDF(URIRef(val.prop_name), content, res_iri)
-
-
-def _transform_file_value(val: AbstractFileValueDeserialised) -> FileValueRDF | None:
+    g = Graph()
+    val_iri = DATA[str(uuid4())]
+    g.add((val_iri, RDF.type, KNORA_API.ListValue))
+    g.add((val_iri, API_SHAPES.listNodeAsString, Literal(node_name, datatype=XSD.string)))
+    g.add((val_iri, API_SHAPES.listNameAsString, Literal(val.list_name, datatype=XSD.string)))
+    g.add((res_iri, URIRef(val.prop_name), val_iri))
+    return g
+
+
+def _transform_file_value(val: AbstractFileValueDeserialised) -> Graph:
+    if val.value is None:
+        return Graph()
     if isinstance(val, IIIFUriDeserialised):
-        return FileValueRDF(
-            res_iri=DATA[val.res_id],
-            value=Literal(val.value, datatype=XSD.anyURI),
-            prop_type_info=IIIF_URI_VALUE,
-            prop_to_value=KNORA_API.stillImageFileValueHasExternalUrl,
-        )
-    return _map_into_correct_file_value(val)
-
-
-def _map_into_correct_file_value(val: AbstractFileValueDeserialised) -> FileValueRDF | None:
+        return _make_file_value_graph(val, IIIF_URI_VALUE, KNORA_API.stillImageFileValueHasExternalUrl)
+    try:
+        file_type = get_file_type_info(val.value)
+        return _make_file_value_graph(val, file_type)
+    except BaseError:
```

Comment:
During the `xmlupload` we stop the resource creation if a wrong extension is used.